elastic/elastic-stack-testing
# Stack Monitoring parity test for Beats

This PR introduces Stack Monitoring parity tests for Beats. Specifically, it introduces a playbook that will install Filebeat and setup internal collection for it. Then it will monitor the same Filebeat instance with external (Metricbeat) collection. Finally, it will compare the internally-and externally-collected monitoring documents for structural parity.

### Testing this PR

1. Visit https://internal-ci.elastic.co/view/Release/job/elastic+release-manager+master+unified-snapshot/lastSuccessfulBuild/console, and note the URL for the latest successful snapshot build. The URL will look like `https://snapshots.elastic.co/MAJOR.MINOR.PATH-SHA_`.

2. Tell the parity tests to use the latest release snapshot. Edit `playbooks/monitoring/buildenv.sh` and set `ES_BUILD_URL` to the URL you noted in step 1. For example:
   ```
   export ES_BUILD_URL=https://snapshots.elastic.co/8.0.0-ae5632f0
   ```

3. Run the monitoring parity tests for Beats.
   ```
   AIT_STACK_PRODUCT=beat ./playbooks/monitoring/buildenv.sh
   ```

5. Verify that the tests pass.

4. Verify that docs of `type: beats_stats` and `type: beats_state` were collected both internally and by Metricbeat. You may need to install the [`tree`](http://mama.indstate.edu/users/ice/tree/) utility first.
   ```
   tree ait_workspace
   ```
